yarl/_url.py

```diff
@@ -1025,6 +1026,16 @@ def update_query(self, *args, **kwargs):
             self._val._replace(query=self._get_str_query(query) or ""), encoded=True
         )
 
+    def without_query_params(self, *query_params: str) -> "URL":
+        """Remove some keys from query part and return new URL."""
+        params_to_remove = set(query_params) & set(self.query.keys())
+        if not params_to_remove:
+            return URL(self._val._replace(query=self.query_string), encoded=True)
+        editable_query = cast(MultiDict, self.query.copy())
+        for k in params_to_remove:
+            editable_query.pop(k, None)
+        return self.with_query(editable_query)
```

Comment:
```suggestion
        return self.with_query(
            (query_param_name, query_param_value)
            for query_param_name, query_param_value in self.query.items()
            if query_param_name not in params_to_remove
        )
```

Reply:
Updated with list comprehension. I use `tuple` as intermediate data, because `with_query()` does not accept `generator` as your code suggested.